File: EventBa.Services/Database/Ticket.cs
using EventBa.Model.Enums;

namespace EventBa.Services.Database;

public partial class Ticket
{
    public Guid Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public Guid EventId { get; set; }
    public TicketType TicketType { get; set; }
    public decimal Price { get; set; }
    public int Quantity { get; set; }
    public int QuantityAvailable { get; set; }
    public int QuantitySold { get; set; }
    public DateTime SaleStartDate { get; set; }
    public DateTime SaleEndDate { get; set; }
    public virtual Event Event { get; set; } = null!;
    public virtual ICollection<TicketPurchase> TicketPurchases { get; set; } = new List<TicketPurchase>();
}

File: UI/eventba_admin/lib/models/ticket/ticket.dart
import 'package:json_annotation/json_annotation.dart';

part 'ticket.g.dart';

@JsonSerializable()
class Ticket {
  final String id;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String eventId;
  final String ticketType;
  final double price;
  final int quantity;
  final int quantityAvailable;
  final int quantitySold;
  final DateTime saleStartDate;
  final DateTime saleEndDate;

  Ticket({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    required this.eventId,
    required this.ticketType,
    required this.price,
    required this.quantity,
    required this.quantityAvailable,
    required this.quantitySold,
    required this.saleStartDate,
    required this.saleEndDate,
  });

  factory Ticket.fromJson(Map<String, dynamic> json) => _$TicketFromJson(json);

  Map<String, dynamic> toJson() => _$TicketToJson(this);
}
